src/displayapp/screens/WatchFacePrideFlag.cpp

```diff
@@ -0,0 +1,321 @@
+  constexpr lv_color_t lightPurple = LV_COLOR_MAKE(0xd4, 0x61, 0xa6);
+  constexpr lv_color_t darkPurple = LV_COLOR_MAKE(0xb5, 0x56, 0x90);
+  constexpr lv_color_t magenta = LV_COLOR_MAKE(0xa5, 0x00, 0x62);
+  constexpr lv_color_t darkGreen = LV_COLOR_MAKE(0x07, 0x8d, 0x70);
+  constexpr lv_color_t cyan = LV_COLOR_MAKE(0x26, 0xce, 0xaa);
+  constexpr lv_color_t lightGreen = LV_COLOR_MAKE(0x98, 0xe8, 0xc1);
+  constexpr lv_color_t indigo = LV_COLOR_MAKE(0x50, 0x49, 0xcc);
+  constexpr lv_color_t steelBlue = LV_COLOR_MAKE(0x3d, 0x1a, 0x78);
+  constexpr std::array<lv_color_t, 7> gayColours {darkGreen, cyan, lightGreen, LV_COLOR_WHITE, lightBlue, indigo, steelBlue};
+  constexpr std::array<lv_color_t, 5> transColours {lightBlue, lightPink, LV_COLOR_WHITE, lightPink, lightBlue};
+  constexpr std::array<lv_color_t, 5> biColours {hotPink, hotPink, grayPurple, darkBlue, darkBlue};
+  constexpr std::array<lv_color_t, 7> lesbianColours {LV_COLOR_RED, orange, lightOrange, LV_COLOR_WHITE, lightPurple, darkPurple, magenta};
+  constexpr WatchFacePrideFlag::PrideFlagData<7> gayFlagData =
+    WatchFacePrideFlag::PrideFlagData(gayColours, LV_COLOR_BLACK, LV_COLOR_BLACK, LV_COLOR_WHITE);
+  constexpr WatchFacePrideFlag::PrideFlagData<5> transFlagData =
+    WatchFacePrideFlag::PrideFlagData(transColours, LV_COLOR_WHITE, LV_COLOR_BLACK, LV_COLOR_WHITE);
+  constexpr WatchFacePrideFlag::PrideFlagData<5> biFlagData =
+    WatchFacePrideFlag::PrideFlagData(biColours, LV_COLOR_BLACK, LV_COLOR_WHITE, LV_COLOR_BLACK);
+  constexpr WatchFacePrideFlag::PrideFlagData<7> lesbianFlagData =
+    WatchFacePrideFlag::PrideFlagData(lesbianColours, LV_COLOR_WHITE, LV_COLOR_BLACK, LV_COLOR_WHITE);
+}
+
+WatchFacePrideFlag::WatchFacePrideFlag(Controllers::DateTime& dateTimeController,
+                                       const Controllers::Battery& batteryController,
+                                       const Controllers::Ble& bleController,
+                                       Controllers::NotificationManager& notificationManager,
+                                       Controllers::Settings& settingsController,
+                                       Controllers::MotionController& motionController)
+  : currentDateTime {{}},
+    dateTimeController {dateTimeController},
+    batteryController {batteryController},
+    bleController {bleController},
+    notificationManager {notificationManager},
+    settingsController {settingsController},
+    motionController {motionController} {
+
+  bluetoothStatus = lv_label_create(lv_scr_act(), nullptr);
+  lv_label_set_text_static(bluetoothStatus, "");
+  lv_obj_align(bluetoothStatus, nullptr, LV_ALIGN_IN_TOP_RIGHT, -16, 0);
+
+  batteryValue = lv_label_create(lv_scr_act(), nullptr);
+  lv_label_set_recolor(batteryValue, true);
+  lv_label_set_align(batteryValue, LV_LABEL_ALIGN_CENTER);
+  lv_obj_set_auto_realign(batteryValue, true);
+
+  notificationIcon = lv_label_create(lv_scr_act(), nullptr);
```

Comment:
I haven't tested this on hardware and it's difficult to do with InfiniEmu, does the notification text look OK on all of the flags?

Also, maybe it'd be worth renaming this, as it's text now rather than an icon